lib/compiler-llvm/src/config.rs

```diff
@@ -207,8 +207,8 @@ impl CompilerConfig for LLVM {
     }
 
     /// Transform it into the compiler.
-    fn compiler(&self) -> Box<dyn Compiler + Send> {
-        Box::new(LLVMCompiler::new(&self))
+    fn compiler(self: Box<Self>) -> Box<dyn Compiler + Send> {
```

Comment:
Consider using `From`/`Into` traits here; not sure how this is used but that could be a bit cleaner

Reply:
I tried, couldn't make it work with `From`. We can do it in a PR apart if necessary